Minimum viable product support for media in Telegram

Lunaphied opened the pull request:
One major downside of the Telegram bot at the moment is not supporting any proxying of messages containing media. Telegram makes messages containing media work in several ways and renames the main body to a "caption", which is exposed in the API as well, meaning that some of the code has to adjust to understand this change.

The "minimum viable product" part of this is that media groups (any message with more than one photo, video, or both), are not supported. These are a bit weirder to support and to do so correctly requires a bit more effort. In the meantime, having anything is better than nothing.

Qyriad commented:
This PR's commits includes #5, which is fine, but it also includes #6, which we've decided not to merge. I don't mind merging #5 in with this (…even if the merge commit message doesn't reflect that), but could you rebase this and remove 68ad9aeffeac94b1c8de2c55b52c5b068a5c4cf0? After that I'd be happy to merge.

Lunaphied commented:
We've updated this to drop that commit. I've also realized that given certain behavior of Telegram we've noticed from testing the media proxying, it's likely that Telegram is actually sending individual messages for each image and then merging them in UI (I could look at the source to confirm this I suppose), which means that it might not be possible to (always anyway) capture the full set of images to proxy as they might not actually come through with any link between them.

Lunaphied committed:
setup: The telegram bot is also part of this package

Lunaphied committed:
telegram: Add MVP support for media

This adds support for messages that contain a single photo or video
without supporting the full media group functionality.

Lunaphied commented:
This has been rebased and should be ready for a merge.
